src/kustomer/webChat/client/src/kustomerServiceDesk.ts

```diff
@@ -0,0 +1,250 @@
+   * so it can perform a more specific check.
+   * @returns True if some agents are available or false if no agents are available. This may also return null which
+   * means the availability status of agents is unknown or the service desk doesn't support this information.
+   */
+  async areAnyAgentsOnline(connectMessage: MessageResponse): Promise<boolean | null> {
+    await this.ensureInit();
+    return KustomerCore.isChatAvailable().availability === 'online';
+  }
+
+  private onMessageReceivedHandler = (response: OnMessageSentResponse, error: any) => {
+    if (response.direction === 'out') {
+      if (!this.agentProfile.nickname) {
+        this.agentProfile.nickname = response.sentBy.displayName;
+        this.agentProfile.id = response.sentBy.id;
+        this.callback.agentJoined(this.agentProfile);
+      } else if (this.agentProfile.nickname !== response.sentBy.displayName) {
+        const newAgentProfile = {
+          nickname: response.sentBy.displayName,
+          id: response.sentBy.id,
+        };
+        this.callback.beginTransferToAnotherAgent(newAgentProfile);
+        this.callback.agentJoined(newAgentProfile);
+      }
+      this.callback.sendMessageToUser(stringToMessageResponseFormat(response.body), this.agentProfile.id);
+    }
+  };
+
+  private onAgentTypingActivity = (response: OnAgentTypingActivityResponse, error: any) => {
+    if (response?.typing === true) {
+      this.callback.agentTyping(true);
+    } else if (response?.typing === false) {
+      this.callback.agentTyping(false);
+    }
+  };
+
+  private onConversationEndedHandler = (response: OnConversationEndedResponse, error: any) => {
+    if (response && response.ended === true) {
+      this.callback.agentEndedChat();
+    }
+  };
+
+  private async initChat(): Promise<void> {
+    return new Promise((resolve, reject) => {
+      KustomerCore.createConversation(
+        {
+          title: CONVERSATION_TITLE,
+        },
+        (response: OnConversationCreateResponse, error: any) => {
+          if (error) {
+            reject(error);
+          } else {
+            const messageObj = {
+              conversationId: response.conversationId,
+              body: WATSON_TRANSFER_MESSAGE,
+            };
+            this.conversationId = response.conversationId;
+            KustomerCore.sendMessage(messageObj, (response: OnSendMessageResponse, error: any) => {
+              if (error) {
+                reject(error);
+              } else {
+                KustomerCore.describeConversation(
+                  {
+                    conversationId: response.conversationId,
+                    customAttributes: {
+                      watsonAssistantSessionHistoryStr: this.sessionHistoryKey,
+                    },
+                  },
+                  (response: any, error: any) => {
+                    if (error) {
+                      reject(error);
```

Comment:
You might want to consider removing this as a `reject`. If this operation fails, we probably shouldn't fail the entire startChat. It's not ideal if the agent doesn't get the conversation history but we should probably allow the connection to occur anywhere. So might be better to just do a `console.error` instead. And maybe include some text in the error message to look at the README for instructions on how to set this up.

Reply:
Good point, change it to console.error